### PR TITLE
API fixes

### PR DESCRIPTION
**Pull request by Jscott7227**
Updated get and post for recipes
Added update and delete for recipes
Added get ingredients
Added Some of Shawn UI changes
Added Recipe filter bar (Kashaina)
Updated recipe add page
Created recipe edit page (Kashaina)
Added get task API pulls

Recipe ingredients currently not working recipes will be added with no ingredients need to update once the backend route gets created.

Final 2 commits contains lots of spacing changes and the changelog updates you can ignore this one when reviewing.

### CODE REVIEW

**Review by kashaina — COMMENTED** [review]
..

File: src/bakery_manager_mobile/lib/assets/constants.dart [hunk 1]
```diff
@@ -3,9 +3,24 @@ const String recipePageRoute = '/recipes';
...
+const String taskDetailsPageRoute = '/tasks/details';
 const String loginPageRoute = '/login';
 const String registrationPageRoute = '/login/registration';
 const String addRecipePageRoute = '/recipes/addRecipe';
+const String editRecipePageRoute = '/recipes/editRecipe';
 const String recipeDetailsPageRoute = '/recipes/details';
 const String myAccountRoute = '/myaccount';
+const String ingredientDetailsPageRoute = "/ingredients/details";
+const String editAccountPageRoute = "/myaccount/edit";
+final List<String> recipeCatagories = [
```
Comment (kashaina) [inline src/bakery_manager_mobile/lib/assets/constants.dart:14]: Are categories finalized?

File: src/bakery_manager_mobile/lib/pages/account_registration.dart [hunk 2]
```diff
@@ -9,68 +9,206 @@ class CreateAccountPage extends StatefulWidget {
...
+          username,
+          password,
+        );
 
-      if (accountCreated) {
-        // Save credentials to SharedPreferences storing the username and password on device
-        SharedPreferences prefs = await SharedPreferences.getInstance();
-        await prefs.setString('username', username);
-        await prefs.setString('password', password);
+        if (accountResponse['status'] == 'success') {
```
Comment (kashaina) [inline src/bakery_manager_mobile/lib/pages/account_registration.dart:93]: we can comment this out for dev?

File: src/bakery_manager_mobile/lib/pages/ingredient_detail_page.dart [hunk 3]
```diff
@@ -0,0 +1,154 @@
+import 'package:flutter/material.dart';
```
Comment (kashaina) [inline src/bakery_manager_mobile/lib/pages/ingredient_detail_page.dart]: I can't test this but lets hope it works

File: src/bakery_manager_mobile/lib/pages/recipe_page.dart [hunk 4]
```diff
@@ -165,36 +266,37 @@ class _RecipeItem extends StatelessWidget {
...
       },
       child: Container(
-        decoration: BoxDecoration(
-          boxShadow: [
-            BoxShadow(
-              color: Colors.grey.withOpacity(0.5), // Shadow under items
-              spreadRadius: 2,
-              blurRadius: 8,
-              offset: const Offset(0, 4), // Shadow offset
+        decoration: const BoxDecoration(
```
Comment (kashaina) [inline src/bakery_manager_mobile/lib/pages/recipe_page.dart]: Delete

File: src/bakery_manager_mobile/lib/services/api_service.dart [hunk 5]
```diff
@@ -241,4 +498,137 @@ class ApiService {
...
+        };
+      } else {
+        return {'status': 'error', 'reason': 'Failed to check Employee ID'};
+      }
+    } catch (e) {
+      return {'status': 'error', 'reason': 'Network error: $e'};
+    }
+  }
+
+  static Future<Map<String, dynamic>> checkEmailExists(String email) async {
```
Comment (kashaina) [inline src/bakery_manager_mobile/lib/services/api_service.dart]: Imma be honest @Jscott7227 , idk if these works

File: src/bakery_manager_mobile/lib/pages/product_page.dart
Comment (kashaina) [inline src/bakery_manager_mobile/lib/pages/product_page.dart:1]: Delete this page

**Review by kashaina — DISMISSED** [review]

File: src/bakery_manager_mobile/lib/assets/constants.dart (hunk 1 above)
Comment (Jscott7227) [reply src/bakery_manager_mobile/lib/assets/constants.dart:14]: No though these can be changed easily if needed.

File: src/bakery_manager_mobile/lib/pages/product_page.dart
Comment (Jscott7227) [reply src/bakery_manager_mobile/lib/pages/product_page.dart:1]: Removed

File: src/bakery_manager_mobile/lib/services/api_service.dart (hunk 5 above)
Comment (Jscott7227) [reply src/bakery_manager_mobile/lib/services/api_service.dart]: They do not using them as a placeholder for now.

File: src/bakery_manager_mobile/lib/pages/recipe_page.dart (hunk 4 above)
Comment (Jscott7227) [reply src/bakery_manager_mobile/lib/pages/recipe_page.dart]: Removed

File: src/bakery_manager_mobile/lib/pages/account_registration.dart (hunk 2 above)
Comment (Jscott7227) [reply src/bakery_manager_mobile/lib/pages/account_registration.dart:93]: Commented out for now

**Review by kashaina — APPROVED** [review]

**Review by kashaina — APPROVED** [review]

**Pull request merged**